R/sampler.R

```diff
@@ -249,3 +248,51 @@ scale_log_weights <- function(log_weights) {
   }
   list(weights = weights, average = average)
 }
+
+
+pre_data_run <- function(model, data, seeding_func,
```

Comment:
Should this also take a `save_particles` option, and return `particles = NULL` if true? Just thinking about memory use

Reply:
Actually I see that you have effectively done this already. I think it would be clearer and more consistent to make this explicit rather than checking `is.null`

Reply:
Done